projects/spectator/test/defer-block.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { Component } from '@angular/core';
+import { DeferBlockBehavior, DeferBlockState, fakeAsync } from '@angular/core/testing';
+import { createComponentFactory } from '@ngneat/spectator';
+
+describe('DeferBlock', () => {
+  describe('Playthrough Behavior', () => {
+    @Component({
+      selector: 'app-root',
+      template: `
+        <button data-test="button--isVisible" (click)="isVisible = !isVisible">Toggle</button>
+
+        @defer (when isVisible) {
+        <div>empty defer block</div>
+        } ,
+      `,
+      standalone: true,
+    })
+    class DummyComponent {
+      isVisible = false;
+    }
+
+    const createComponent = createComponentFactory({
+      component: DummyComponent,
+      deferBlockBehavior: DeferBlockBehavior.Playthrough,
+    });
+
+    it('should render the defer block when isVisible is true', fakeAsync(() => {
+      // Arrange
+      const spectator = createComponent();
+
+      const button = spectator.query('[data-test="button--isVisible"]')!;
+
+      // Act
+      spectator.click(button);
+      spectator.tick();
+      spectator.detectChanges();
+
+      // Assert
+      expect(spectator.element.outerHTML).toContain('empty defer block');
+    }));
+  });
+
+  describe('Manual Behavior', () => {
+    @Component({
+      selector: 'app-root',
+      template: `
+        @defer (on viewport) {
+        <div>empty defer block</div>
+        } @placeholder {
+        <div>placeholder</div>
+        }
+      `,
+    })
+    class DummyComponent {}
+
+    const createComponent = createComponentFactory({
+      component: DummyComponent,
+      deferBlockBehavior: DeferBlockBehavior.Manual,
+    });
+
+    it('should render the complete state', async () => {
+      // Arrange
+      const spectator = createComponent();
+
+      // Act
+      const deferFixture = (await spectator.fixture.getDeferBlocks())[0];
```

Comment:
What do u think about making the API nicer?

```ts
spectator.deferBlocks.renderComplete();
spectator.deferBlocks.renderPlaceholder();
etc..
```

Reply:
That's a great idea! 
Thanks for that. I'll work on this :)

Reply:
I followed the same API for the nested defer blocks as well.
Please let me know if this is OK and if it is required something else.